packages/expect/src/types.ts

```diff
@@ -325,7 +339,7 @@ export interface Matchers<R extends void | Promise<void>> {
   /**
    * Ensure that a mock function has returned a specified value at least once.
    */
-  toReturnWith(expected: unknown): R;
+  toReturnWith<U extends EnsureFunctionLike<T>>(expected: ReturnType<U>): R;
```

Comment:
Maybe change to this syntax? it works and causes less indirection
```suggestion
  toReturnWith(expected: ReturnType<EnsureFunctionLike<T>>): R;
```

Reply:
Thanks. That’s good idea. I was hoping that type argument could allow something like: `expect(123).toHaveReturnedWith<() => number>(123)`. But it doesn’t.